### PR TITLE
Add filters

### PR DESCRIPTION
**Pull request by pavan4**
- Adds hybrid search 
- refactor code

### CODE REVIEW

File: src/handlers/query.ts [hunk 1]
```diff
@@ -277,6 +269,7 @@ function getSearchTextFilter(
       switch (expression.operator) {
         case "near_text":
         case "match_text":
+        case "hybrid_match_text":
           if (negated) {
             throw new Error("Negated near_text or match_text not supported");
```
Comment (nickthegroot) [inline src/handlers/query.ts]: 🙃 Would be nice to have `hybrid_match_text` included in the error message

(can't comment there but ditto on line 277)

File: src/handlers/query.ts [hunk 2]
```diff
@@ -151,16 +151,22 @@ async function executeSingleQuery(
...
+          concepts: searchTextFilter,
+        });
+      } else if (isTextFilter(query.where, "match_text")) {
+        getter.withBm25({
+          query: searchTextFilter.toString(),
+        });
+      } else if (isTextFilter(query.where, "hybrid_match_text")) {
+        getter.withHybrid({
+          query: searchTextFilter.toString(),
+        });
```
Comment (nickthegroot) [inline src/handlers/query.ts:166]: 🤔 Is there anyway for us to customize the `alpha` parameter of the hybrid search? Not sure we need it with the current plan (semantic & keyword displayed separately), but would be nice to be able to fine-tune.

**Review by nickthegroot — APPROVED** [review]
Nice! LGTM, thanks for doing this :+1:

File: src/config.ts [hunk 3]
```diff
@@ -15,10 +15,10 @@ export const getConfig = (request: FastifyRequest): Config => {
...
   const config = JSON.parse(rawConfigJson);
   return {
-    host: process.env[config.host] ?? config.host,
-    scheme: process.env[config.scheme] ?? config.scheme ?? "http",
-    apiKey: process.env[config.apiKey] ?? config.apiKey,
-    openApiKey: process.env[config.openApiKey] ?? config.openApiKey,
+    host: config.host,
+    scheme: config.scheme ?? "http",
+    apiKey: config.apiKey,
+    openApiKey: config.openApiKey,
```
Comment (nickthegroot) [inline src/config.ts:21]: ❓  Totally fine, but curious why these were removed?

Comment (pavan4) [reply src/config.ts:21]: for better or worse, config is loaded for each query/mutation https://github.com/digs-com/weaviate_gdc/blob/c3cbf9bd37df8b37aa6292c83400639b815212f0/src/index.ts#L61 If the env var is refreshed on hasura side, that value is reflected immediately. If this is set via env var, then any changes to the env var on hasura will never be picked up. So I removed it. Can add it back if needed?

File: src/handlers/query.ts (hunk 2 above)
Comment (pavan4) [reply src/handlers/query.ts:166]: I was planning to handle `alpha` and `fusionType` as a follow up PR.

**Pull request merged**